src/browser-frontend/views/browser/chrome/tabbar/tabs-list.jsx

```diff
@@ -10,37 +10,72 @@ CONDITIONS OF ANY KIND, either express or implied. See the License for the
 specific language governing permissions and limitations under the License.
 */
 
-import React, { PureComponent } from 'react';
+import React, { PureComponent, PropTypes } from 'react';
 import CSSModules from 'react-css-modules';
 import { connect } from 'react-redux';
+import { SortableContainer, SortableElement } from 'react-sortable-hoc';
 
 import * as SharedPropTypes from '../../../../model/shared-prop-types';
 import * as UIPagesSelectors from '../../../../selectors/ui-pages-selectors';
+import PagesModelActions from '../../../../actions/pages-model-actions';
 
 import Styles from './tabs-list.css';
 import Tab from './tab';
 
+const SortableTab = SortableElement(({ pageId }) =>
+  <Tab
+    key={pageId}
+    pageId={pageId}
+  />,
+);
+const SortableList = SortableContainer(({ pageIds }) =>
+  <div>
+    {pageIds.map((pageId, index) => (
+      <SortableTab
+        key={pageId}
+        pageId={pageId}
+        index={index}
+      />
+    ))}
+  </div>,
+);
+
 @connect(state => ({
   pageIds: UIPagesSelectors.getPageIdsInDisplayOrder(state),
 }))
 @CSSModules(Styles, {
   allowMultiple: true,
 })
 export default class TabsList extends PureComponent {
+  onSortEnd = ({ oldIndex, newIndex }) => {
+    const pageIds = this.props.pageIds;
+    if (oldIndex !== newIndex) {
+      const pageId = pageIds._tail.array[oldIndex];
```

Comment:
Accessing private Immutable.js members seems like a clear no-no. Any reason why the public APU couldn't be used?